controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -117,6 +117,11 @@ type KubeadmControlPlaneSpec struct {
 	// The RemediationStrategy that controls how control plane machine remediation happens.
 	// +optional
 	RemediationStrategy *RemediationStrategy `json:"remediationStrategy,omitempty"`
+
+	// MachineNamingStrategy allows changing the naming pattern used when creating Machines.
+	// Note: InfraMachines & KubeadmConfigs will use the same name as the corresponding Machines.
+	// +optional
+	MachineNamingStrategy *MachineNamingStrategy `json:"machineNamingStrategy,omitempty"`
```

Comment:
I think this also has to be added to KubeadmControlPlaneTemplate here: https://github.com/kubernetes-sigs/cluster-api/blob/0b6aa4926213a9058da62643395348f3a94065ba/controlplane/kubeadm/api/v1beta1/kubeadmcontrolplanetemplate_types.go#L103